src/identity.ts

```diff
@@ -0,0 +1,227 @@
+  dateModified: string /** The last date that the identity was modified. */;
+}
+
+/** The results from a getIdentities search. */
+interface IEveryAuthIdentitySearch {
+  items: IEveryAuthIdentity[];
+  next?: string /** A token that can be supplied back to request the next page of results. */;
+}
+
+/** Options that control the pagination of getIdentities requests. */
+export interface IEveryAuthSearchOptions {
+  next?: string;
+  pageSize?: number;
+}
+
+/** A specific credential associated with an identityId, normalized to provide a standard interface. */
+interface IEveryAuthCredential {
+  native: provider.INative /** The raw identity from a service. */;
+  accessToken: string /** The OAuth2 accessToken used to authenticate a request. */;
+}
+
+interface IEveryAuthUserTenantSet {
+  userId?: string;
+  tenantId?: string;
+}
+
+/**
+ * Retrieve a valid accessToken for a specific service.  If search criteria such as a userId or tags are
+ * used, only the most recent matching identity will be returned.
+ *
+ * @param serviceId The service to search for matching identities within.
+ * @param identityOrIdsOrTags Either an identity id uniquely identifying a specific identity, a
+ * userId that can be used to search for a matching identity, or a set of tags that will be used to
+ * search.
+ * @return A credential with a valid token.
+ * @throws An exception is thrown if no identities are found, or more than one identity is found.
+ */
+export const getIdentity = async (
+  serviceId: string,
+  identityOrIdsOrTags: string | IEveryAuthUserTenantSet | IEveryAuthTagSet
+): Promise<IEveryAuthCredential> => {
+  let identityId;
+
+  // Is this already an identity?
+  if (typeof identityOrIdsOrTags == 'string' && identityOrIdsOrTags.startsWith('idn-')) {
+    identityId = identityOrIdsOrTags;
+  } else {
+    let identities: IEveryAuthIdentitySearch;
+
+    if (typeof identityOrIdsOrTags == 'string') {
+      identities = await getIdentitiesByTags(serviceId, { [USER_TAG]: identityOrIdsOrTags });
+    } else if (identityOrIdsOrTags.userId || identityOrIdsOrTags.tenantId) {
+      identities = await getIdentitiesByTags(serviceId, {
+        [USER_TAG]: identityOrIdsOrTags.userId,
+        [TENANT_TAG]: identityOrIdsOrTags.tenantId,
+      });
+    } else {
+      identities = await getIdentitiesByTags(serviceId, identityOrIdsOrTags as IEveryAuthTagSet);
+    }
+
+    debug(`${JSON.stringify(identityOrIdsOrTags)}: Found ${identities.items.length} matching identities`);
+    if (identities.items.length > 1) {
+      throw new Error(
+        `The userId "${JSON.stringify(
+          identityOrIdsOrTags
+        )}" resolves to more than one identity. Either use "getIdentities" to list all of the matching identities, or remove redundant identity using "everynode identity rm" or "deleteIdentity"`
+      );
+    }
+
+    if (identities.items.length == 0) {
+      throw new Error('No credentials found');
```

Comment:
We discussed returning `undefined` in this case, as it is the normal code path through the system - this is how people would check if a particular user of theirs has already authorized a service or not.